components/bl-rating-component/src/index.js

```diff
@@ -0,0 +1,91 @@
+import { useCallback, useRef, useState } from 'react';
+
+import { getIconHoverValue, useRatingClassList } from './helpers';
+import { RatingIcon } from './rating-icon';
+
+const iconsList = iconsAmount => new Array(iconsAmount).fill(1);
+
+export default function RatingComponent({ component, eventHandlers }) {
+  const {
+    display,
+    disabled,
+    icon,
+    defaultValue,
+    iconsAmount,
+    iconColor,
+    precision,
+    classList,
+  } = component;
+  const { onRatingChange } = eventHandlers;
+
+  const rootRef = useRef();
+  const classes = useRatingClassList(disabled, classList);
+  const [ratingValue, setRatingValue] = useState(defaultValue);
+  const [hoverValue, setHoverValue] = useState();
+
+  const icons = iconsList(iconsAmount);
+  const value = hoverValue || ratingValue;
+
+  const handleRatingValue = useCallback(({ target: { value  } }) => {
+    setRatingValue(prevState => prevState === value ? 0: value);
+
+    if (onRatingChange) {
+      const requestParams = {
+        curentRating: value,
+      };
+
+      if (Number(value) === iconsAmount) {
+        requestParams.maxRating = value;
+      }
+
+      if (value === precision) {
+        requestParams.minRating = value;
+      }
+
+      onRatingChange(requestParams);
+    }
+  }, []);
+
+  const handleMouseMove = event => {
+    if (disabled) {
+      return;
+    }
+
+    getIconHoverValue(rootRef.current, iconsAmount, event.clientX, setHoverValue, precision);
```

Comment:
setHoverValue(getIconHoverValue(rootRef.current, iconsAmount, event.clientX, precision))

Also, could we pass the event, and instead `rootRef.current` get currentTarget from event? In this case that could look like below:

`setHoverValue(getIconHoverValue(event, iconsAmount, precision))`